### ISSUE TITLE
wrong newlines when pasting a public key

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#2860]
1) make sure to choose "plain text" type of message using the compose button chevron (this should be the default anyway)
2) copy public key from https://flowcrypt.com/pub/human@flowcrypt.com?show=pubkey
3) paste it into secure compose

result:

![image](https://user-images.githubusercontent.com/6306961/88160685-89aa6a00-cbfe-11ea-8926-d2cd842943d6.png)

expected result:

![image](https://user-images.githubusercontent.com/6306961/88160749-a0e95780-cbfe-11ea-9043-0f1904c81967.png)

I don't know where the problem happens, whether at time of copying or pasting.

**Comment by nomad-dad** [comment]
**Note:** If any FlowCrypt user is having this issue and is eager to be able to paste from `pre` tags in the meantime, there's a simple workaround - just use ctrl+shift+v to paste (instead of the usual ctrl+v)

---

This occurs whenever we paste text copied from inside a `<pre>` tag.

When a user pastes formatted code, FlowCrypt remove all newlines here:

https://github.com/FlowCrypt/flowcrypt-browser/blob/1d7312c29616634fc0c73144fe2424fb36275bf2/extension/js/common/platform/xss.ts#L181

So now, there are no newlines inside of the `pre`. With other tags, this is okay, since the code goes on to make sure to preserve the newlines provided by tags like `div`, `table`, and so on. These tags create two newlines: one above, and one below. 

However, `pre` is a problem, because the newlines _inside_ of the tag are not delineated in any special way.

So we need to convert these newlines in `br`'s, but in a way that _only_ targets newlines inside of `pre` tags (my guess is that replacing *all* newlines with `br`s causes some other issue, hence why https://github.com/FlowCrypt/flowcrypt-browser/pull/3001 was rejected).

There are myriad ways to solve this problem, now that we've identified what the problem actually is. The simplest is to add another regex that replaces `\n` with `br` _if we're inside of a `pre`_, but I'd prefer to use the browser's built-in [DOMParser](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser) utility to avoid spending lots of time debugging a regex.

```javascript
const messageDomParser = new DOMParser();
const messageDom = messageDomParser.parseFromString(html, 'text/html');
const preTags = messageDom.getElementsByTagName('pre');
for (const pre of preTags) {
    pre.innerHTML = pre.innerHTML.replace(/\n/g, br);
}
html = messageDom.body.innerHTML;
html = html.replace(/\n/g, '');
```

So in other words, what @wolmir did but only inside of `pre`, and using DOMParser instead of regex (because I'm rusty with regex - but it wouldn't be too hard to do it that way if this is too verbose).

Let's see the result:

<img width="676" alt="Screenshot 2023-06-14 at 15 48 50" src="https://github.com/FlowCrypt/flowcrypt-browser/assets/104947308/4f39cdce-c704-4b21-a5bc-85a95acf7520">

It works, we can now paste from `pre` tags. I made a PR for this, should be linked automatically.
